src/shared/sol-conffile.c

```diff
@@ -58,87 +60,207 @@ sol_conffile_entry_sort_cb(const void *data1, const void *data2)
-    GError *error = NULL;
+    const char *include_str = "Include";
+    const char *include_fallback = "IncludeFallbacks";
 
-    if (!g_key_file_has_group(keyfile, include_group))
-        return;
+    bool found_include = false;
 
-    *include = g_key_file_get_string(keyfile,
-        include_group, "Include", &error);
-    if (*include == NULL && error
-        && error->code != G_KEY_FILE_ERROR_KEY_NOT_FOUND) {
-        SOL_DBG("could not read value from %s group: %s",
-            include_group, error->message);
-        g_error_free(error);
-    }
+    struct sol_json_token token, key, value;
+    struct sol_json_scanner include_scanner;
+    enum sol_json_loop_reason reason;
 
-    *include_fallbacks = g_key_file_get_string(keyfile, include_group,
-        "IncludeFallbacks", &error);
-    if (*include_fallbacks == NULL && error
-        && error->code != G_KEY_FILE_ERROR_KEY_NOT_FOUND) {
-        SOL_DBG("could not read value from %s group: %s",
-            include_group, error->message);
-        g_error_free(error);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&json_scanner, &token, &key, &value, reason) {
+        if (reason != SOL_JSON_LOOP_REASON_OK) {
+            SOL_DBG("Error: Invalid Json.");
+            break;
+        }
+
+        if (sol_json_token_str_eq(&key, include_group, strlen(include_group))) {
+            found_include = true;
+            break;
+        }
     }
 
-    if (!g_key_file_remove_group(keyfile, include_group, &error)) {
-        SOL_DBG("could not remove %s group: %s", include_group,
-            error ? error->message : "unknown error");
-        g_error_free(error);
+    if (!found_include)
+        return;
+
+    sol_json_scanner_init_from_token(&include_scanner, &value);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&include_scanner, &token, &key, &value, reason) {
+        if (reason != SOL_JSON_LOOP_REASON_OK) {
+            SOL_DBG("Error: Invalid Json.");
+            break;
+        }
+        if (sol_json_token_str_eq(&key, include_str, strlen(include_str))) {
+            *include = strndup(value.start, sol_json_token_get_size(&value));
+            if (!*include)
+                SOL_DBG("Error: couldn't allocate memory for string.");
+            continue;
+        }
+        if (sol_json_token_str_eq(&key, include_fallback, strlen(include_fallback))) {
+            *include_fallbacks = strndup(value.start, sol_json_token_get_size(&value));
+            if (!*include_fallbacks)
+                SOL_DBG("Error: couldn't allocate memory for string.");
+            continue;
+        }
     }
 }
 
-static GKeyFile *
-sol_conffile_load_keyfile_from_dirs(const char *file, char **full_path)
+static struct sol_str_slice
+sol_conffile_load_json_from_dirs(const char *file, char **full_path, struct sol_file_reader **file_reader)
 {
-    char *cwd = NULL;
-    GError *error = NULL;
-    GKeyFile *keyfile = NULL;
-    char cwdbuf[PATH_MAX + 1];
+    size_t i;
+    struct sol_str_slice config_file_contents;
```

Comment:
Initialize this slice with an empty slice (there's a macro for that in the slice include). This is because, if the `else` below fails, `config_file_contents` will be used uninitialized.